### PR TITLE
Final check before XMC1400 BSP/ CAN Lib release

### PR DESCRIPTION
**Pull request by LinjingZhang**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Details in https://jirard.intra.infineon.com/browse/DESMAKERS-3943
code cleaned , doc updated. example tested. 

Related Issue
https://jirard.intra.infineon.com/browse/DESMAKERS-3943

Context
I added the unity structure I had before.
@jaenrig-ifx  I changed the release script a bit, so the test files and make files will not be included in release. And make is available for CICD :D

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+# We'll use defaults from the LLVM style, but with 4 columns indentation.
```
Comment (jaenrig-ifx) [inline .clang-format:2]: How to run this?
While this is not built-in automated check in the project, at least, we should add some documentation on how to locally run this formatter or check the format.
The truth is that without ci/cd check, the formatting won´t be ensured for every contribution. And if that is not the case... this is yet not very useful.

File: libraries/CAN/Makefile.test [hunk 2]
```diff
@@ -2,12 +2,23 @@
 
 
 ### Example program
+## XMC4700_Relax_Kit
```
Comment (jaenrig-ifx) [inline libraries/CAN/Makefile.test:5]: How to run this? 
Where is the makefile? etc?

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
Maybe we can document briefly those aspects of the formatting and the tests?
The CAN functional changes I have no judgment criteria. I expect it has been verified in the best way possible (manually).

File: .clang-format (hunk 1 above)
Comment (LinjingZhang) [reply .clang-format]: .clang_format removed!

File: libraries/CAN/Makefile.test (hunk 2 above)
Comment (LinjingZhang) [reply libraries/CAN/Makefile.test:5]: I will keep this for now

File: .github/scripts/release.py [hunk 3]
```diff
@@ -37,8 +37,12 @@ def build_package(pkg_name):
         'README.md'
     ]
 
+    ignore_pattern = shutil.ignore_patterns()
+
     for dir in dirs_to_copy:
-        shutil.copytree(os.path.join(xmc_ino_root_path, dir), os.path.join(pkg_build_path, dir))
```
Comment (jaenrig-ifx) [inline .github/scripts/release.py:41]: Increase the version of the script to 0.2.2?

**Review by jaenrig-ifx — APPROVED** [review]

File: .github/scripts/release.py (hunk 3 above)
Comment (LinjingZhang) [reply .github/scripts/release.py:41]: The current version in master is 0.2.0. So 0.2.1 should be correct 🥲 But I don't know why the git log is so strange. We revert this file some times a go....
But in final merge you will see the script is correctly versioned

Comment (jaenrig-ifx) [reply .github/scripts/release.py:41]: if that is the case 👍

**Pull request merged**